src/mame/handheld/hh_ht11xx.cpp

```diff
@@ -0,0 +1,118 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+
+
+#include "emu.h"
+
+#include "cpu/ht1130/ht1130.h"
+
+#include "screen.h"
+#include "speaker.h"
+
+#define VERBOSE (0)
+#include "logmacro.h"
+
+namespace {
+
+class ht11xx_brickgame_state : public driver_device
+{
+public:
+	ht11xx_brickgame_state(const machine_config &mconfig, device_type type, const char *tag) :
+		driver_device(mconfig, type, tag),
+		m_maincpu(*this, "maincpu"),
+		m_out_x(*this, "seg%u_%u", 0U, 0U),
+		m_in1(*this, "IN1"),
+		m_in2(*this, "IN2")
+	{ }
+
+	void ht11xx_brickgame(machine_config &config);
+
+	DECLARE_INPUT_CHANGED_MEMBER(power_pressed);
+
+protected:
+	virtual void machine_start() override;
+	virtual void machine_reset() override;
+
+private:
+	void mcfg_svg_screen(machine_config &config, u16 width, u16 height, const char *tag = "screen");
+
+	void display_data_w(offs_t offset, u8 data);
+
+	required_device<ht1130_device> m_maincpu;
+	output_finder<256, 4> m_out_x;
+	required_ioport m_in1;
+	required_ioport m_in2;
+};
+
+void ht11xx_brickgame_state::machine_start()
+{
+	m_out_x.resolve();
+}
+
+void ht11xx_brickgame_state::machine_reset()
+{
+	for (int i=0;i<256;i++)
+		for (int j=0;j<4;j++)
+			m_out_x[i][j] = 0;
```

Comment:
Is this actually correct?  The datasheet for the CPU says, “All LCD segments are random after an initial clear.”  (See LCD driver output section on page 12.)  This suggests that a reset doesn’t affect the contents of display memory or the LCD outputs, and software shouldn’t expect the outputs to be in a given state on start.

At any rate, if you’re keeping it, please put some spaces in the controlling conditions for the loops.  (You could also use range-based `for` loops.)

Reply:
probably not, I was using it when converting the layout, to make sure I'd named all the segments properly.